app/controllers/concerns/foreman_remote_execution/hosts_controller_extensions.rb

```diff
@@ -0,0 +1,36 @@
+module ForemanRemoteExecution
+  module HostsControllerExtensions
+    extend ActiveSupport::Concern
+    include Foreman::Renderer
+
+    included do
+      alias_method_chain(:action_permission, :remote_execution)
+    end
+
+    def reprovision
+      find_resource
+      script_template = @host.provisioning_template(:kind => 'script')
+      if script_template.nil?
+        process_error :redirect => :back, :error_msg => _("No script provisioning template available")
+        return
+      end
+      @host.setBuild
+      script = unattended_render(script_template.template, @template_name)
+      composer = JobInvocationComposer.for_feature(:reprovision, @host, :script => script)
+      composer.save!
+      composer.trigger
```

Comment:
Yea, that would work.  It's really not a big deal, I just noticed we keep making minor changes to how this gets called and figured maybe it could be DRYer.